crates/re_data_source/src/data_loader/loader_external.rs

```diff
@@ -185,6 +190,39 @@ impl crate::DataLoader for ExternalLoader {
                     }
                 };
 
+                // We have to wait in order to know whether the child process is a compatible loader.
+                //
+                // This can manifest itself in two distinct ways:
+                // 1. If it exits immediately with an INCOMPATIBLE exit code, then we have our
+                //   answer straight away.
+                // - If it starts streaming data, then we immediately assume it's compatible.
+                loop {
+                    match child.try_wait() {
+                        Ok(Some(_)) => break,
+                        Ok(None) => {
+                            // The child process has started streaming data, it is therefore compatible.
+                            // Let's get out ASAP.
+                            if rx_is_sending_data.try_recv().is_ok() {
```

Comment:
nit: we don't know that the child process has streamed data until we're inside the nit (I was confused for a few seconds thinking `Ok(None)` meant the child process had started streaming data)

```suggestion
                            if rx_is_sending_data.try_recv().is_ok() {
                                // The child process has started streaming data, it is therefore compatible.
                                // Let's get out ASAP.
```